pkg/apis/broker/v1beta1/broker_lifecycle.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"knative.dev/eventing/pkg/apis/duck"
+	"knative.dev/pkg/apis"
+)
+
+var brokerCondSet = apis.NewLivingConditionSet(
+	BrokerConditionTopic,
+	BrokerConditionSubscription,
+	BrokerConditionIngress,
+	BrokerConditionAddressable,
+)
+
+const (
+	BrokerConditionReady                           = apis.ConditionReady
+	BrokerConditionTopic        apis.ConditionType = "TopicReady"
+	BrokerConditionSubscription apis.ConditionType = "SubscriptionReady"
+	BrokerConditionIngress      apis.ConditionType = "IngressReady"
+	BrokerConditionAddressable  apis.ConditionType = "Addressable"
+)
+
+// GetCondition returns the condition currently associated with the given type, or nil.
+func (bs *BrokerStatus) GetCondition(t apis.ConditionType) *apis.Condition {
+	return brokerCondSet.Manage(bs).GetCondition(t)
+}
+
+// GetTopLevelCondition returns the top level Condition.
+func (bs *BrokerStatus) GetTopLevelCondition() *apis.Condition {
+	return brokerCondSet.Manage(bs).GetTopLevelCondition()
+}
+
+// IsReady returns true if the resource is ready overall.
+func (bs *BrokerStatus) IsReady() bool {
+	return brokerCondSet.Manage(bs).IsHappy()
+}
+
+// InitializeConditions sets relevant unset conditions to Unknown state.
+func (bs *BrokerStatus) InitializeConditions() {
+	brokerCondSet.Manage(bs).InitializeConditions()
+}
+
+func (bs *BrokerStatus) MarkTopicFailed(reason, format string, args ...interface{}) {
+	brokerCondSet.Manage(bs).MarkFalse(BrokerConditionTopic, reason, format, args...)
```

Comment:
Somewhere in this file we probably should comment that these status are addition to "standard" broker status?